582 enhancement pull request template (#583)
DLR-SC/memilio: .github/pull_request_template.md

## Changes

Please briefly list the changes made.

## Merge Request - GuideLine Checklist

> Please check our [git workflow](https://github.com/DLR-SC/memilio/wiki/git-workflow) before opening a PR and request a reviewer when this is ready to review. If this is Work in Prorgess please use the draft feauture.

Author: This is cleaner and better

### Checks by code author

> * [ ] Every addressed Issue is linked (use the "Closes #ISSUE" keyword)

Reviewer: ```suggestion
* [ ] Every addressed issue is linked (use the "Closes #ISSUE" keyword below this template)
```

Author: As i want to avoid as much "fluff" i just added the below

* [ ] New code adheres to [coding guidelines](https://github.com/DLR-SC/memilio/wiki/Coding-guidelines)
* [ ] No large data files added (max size in KB, avoid PDFs, Word docs, etc.)

> * [ ] Tests are added for new functionality and run successful

Reviewer: ```suggestion
* [ ] Tests are added for new functionality and a local test run was successful
```

Author: Better

> * [ ] Appropriate **documentation** for new functionality (doxygen)

Reviewer: ```suggestion
* [ ] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
```

> * [ ] Proper license attention for third-party software

Reviewer: ```suggestion
* [ ] Proper attention to licenses: No new third-party software with conflicting license has been added
```

Author: yes, added an abbreviated version of the suggested change

### Checks by code reviewer(s)

> * [ ] The issue is addressed

Reviewer: ```suggestion
* [ ] Corresponding issue(s) is/are linked and addressed
```

Author: Way better

* [ ] Clean code of development artifacts (unnecessary comments, prints,large data files, etc.)

> * [ ] Appropriate and passing **unit tests** with acceptable coverage report

Reviewer: ```suggestion
* [ ] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
```

> * [ ] No large data files added (max size in KB, avoid PDFs, Word docs, etc.)
> 

Reviewer: ```suggestion
* [ ] No large data files have been added in the whole history of commits (by default, files should not exceed 100 KB, new files should not sum up to MBs, avoid PDFs, Word docs, etc.)

```
